src/java.base/share/classes/java/net/URL.java

```diff
@@ -554,12 +575,19 @@ public URL(String spec) throws MalformedURLException {
      * <p>
      * For a more detailed description of URL parsing, refer to RFC2396.
      *
+     * @implSpec Parsing the URL involves calling the {@link
+     * URLStreamHandler#parseURL(URL, String, int, int) parseURL} method on the
+     * selected handler.
```

Comment:
What you would think about making this more direct, e.g. "This constructor invokes the selected stream handler's parseURL method to parse the URL."

Reply:
Well some of the parsing is done in URL itself: like finding the fragment, if any, setting up start and limit etc... And I didn't want to get into the rabbit hole of trying to specify what does what exactly. I can still take your suggested text if you believe it doesn't matter?

Reply:
Okay, maybe your wording would be be okay if you changed "involves" to "includes" as the former hints that there is more going on that is not specified.

Reply:
Good idea! Will do.